### ISSUE TITLE
Searching after filtering for a Category maintains the category in the URL & header

### ISSUE THREAD

**Issue by lupus2k** [Automattic/wp-calypso#66495]
#### Description
When filtering for a category and then searching for a text (or clicking on a popular search), the header shows the Category & Search Results, and the URL uses both, Category and search text.
We are not searching within a category, we have to remove the category when searching.

#### Steps to reproduce

1. Go to `/plugins`
2. Click on a category (ex. `Search Engine Optimization`)
3. Click on a popular search (or search something)

#### What I expected
* In page header, it should show `Plugins > Search Results`
* The URL should be `/plugins/:siteId:?s=searchtext`

#### What happened instead
* Page header shows the `Plugins > :Category: > Search Results`
* The URL shows `/plugins/browse/seo/:siteid:?s=searchtext`

#### Screenshot / Video
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/402286/184129465-75a5999f-b86d-441f-948a-5b5343e67ce4.png">

#### Context / Source
In Marketplace, we are not searching within categories.

**Comment by lsl** [comment]
Thanks for creating an issue for this. Can we fix this by implementing category search with the ES integration?

Acceptance criteria for that might look like:
- [ ] Add the `categoryName` prop back to the [PluginSearchResultPage](https://github.com/Automattic/wp-calypso/blob/f4e1c9e3a044e17eef1e22d58fce30dbb54548af/client/my-sites/plugins/plugins-browser/index.jsx#L198) component props (so the text reads `found x plugins for "search" under "category"`
- [ ] Limit search results by selected category

**Comment by lupus2k** [comment]
Header path is fixed in #66668

**Comment by lupus2k** [comment]
#### When filtering for a Category and then searching.
- [ ] It maintains the category on Header's path. (Fixes in #66668)
- [ ] It maintains the category on the URL. (Ex: `/plugins/browse/seo?s=ecommerce`)

#### When filtering for a Category, then search, and then clear the search with the Clear button on the plugin's list.
- [ ] It changes the URL to `/plugins` but keeps the category filter. (see screenshot)
- [ ] - [ ] It changes the URL to `/plugins` but keeps the category on Header's path. (see screenshot)
- [ ] It clears the search, but the search input keeps the text (see screenshot) (Fixes in #66668)

#### Screenshot
![image](https://user-images.githubusercontent.com/402286/185498032-feb2bd19-b3c5-42dc-9e24-a23083d457ce.png)

**Comment by lupus2k** [comment]
> Thanks for creating an issue for this. Can we fix this by implementing category search with the ES integration?

Sounds good, but I think the ES integration could take us longer, and these are quick fixes.

What do you think about creating a new issue with the ES integration for search?

**Comment by lsl** [comment]
The urls, routing and navigation clicks in the category selector are already working how we need it to work for category filtered search right now. If we are going to change any of that I think we just need to make sure it's easy to revert back in a week or two. For example I want to avoid [this sort of change](https://github.com/Automattic/wp-calypso/pull/66668/files?w=1#r949730282) because we'll only need to bring it back once ES is done.

ES integration with calypso is already semi deployed behind a flag, I think we just need to hook up category searches, see the `marketplace-jetpack-plugin-search` flag in `usePlugins`.

e.g. https://wordpress.com/plugins/?flags=marketplace-jetpack-plugin-search

The blocker for deploying ES for everyone is just a wccom sync or something like that but I think it's being worked on this sprint by bespin.

**Comment by lupus2k** [comment]
Thanks, @lsl, for the update on this!

**Comment by lsl** [comment]
@lupus2k I want to reverse my thinking on this, I think you're right, lets fix the behavior and revisit later. I'm thinking making decisions on code behind a feature flag is a bad choice and this was the correct thinking:

> Sounds good, but I think the ES integration could take us longer, and these are quick fixes.

Can we do this in a way that's easy to modify later if we need to? (I don't think its a blocker but that would be a nice side benefit if it doesn't add any scope)

**Comment by lupus2k** [comment]
I'm gonna close this issue because we fix almost all errors on refactors, in #66883, and new issue created about `Close` issue: #66938
